### PR TITLE
Fix default paths for `rvm-auto-ruby`

### LINKED ISSUE #1862: Failed to activate rvm environment
### Description

I've been using the Ruby LSP vscode extension for about a year without any issues. 

Today I started seeing the error message below when I launch vscode. I'm launching vscode from the terminal with the `code .` command.
```
Failed to activate rvm environment: Command failed: /home/onwardmk/.rvm/bin/rvm-auto-ruby -W0 -rjson -e 'STDERR.print({yjit:!!defined?(RubyVM::YJIT),version:RUBY_VERSION,home:Gem.user_dir,default:Gem.default_dir,ruby:RbConfig.ruby}.to_json)' /bin/sh: 1: /home/onwardmk/.rvm/bin/rvm-auto-ruby: not found
```

Based on the error message, the problem appears to be that it's looking for rvm in my home folder, but rvm is installed in `/usr/share/rvm` on my machine. I don't see a vscode setting to specify the path.

vscode version: 1.87.2
Ruby LSP version: v0.5.16
rvm version: 1.29.12

### PR DESCRIPTION
**Pull request by joennlae**
First of all, this is a fantastic project 💯 . While setting it up today, I found this issue (#1862) and saw that it was probably introduced recently.

This commit not only checks the path:

`/home/user/.rvm/bin` (done before)

It also checks if it is in the PATH already and in: `/usr/local/rvm/bin`

It additionally adds some output if the initialization fails.

### Motivation

Closes #1862

### Implementation

It should be self-explanatory.

### Automated Tests

I added no additional test, but I tested it on my machine (make of what you want :P)

### Manual Tests

I tested it on my machine by installing the extension via the `vsce package`

### CODE REVIEW

**Comment by joennlae** [comment]
I have signed the CLA!

File: vscode/src/ruby/rvm.ts [hunk 1]
```diff
@@ -19,13 +19,53 @@ export class Rvm extends VersionManager {
...
-    const result = await asyncExec(
-      `${path.join(os.homedir(), ".rvm", "bin", "rvm-auto-ruby")} -W0 -rjson -e '${activationScript}'`,
-      {
-        cwd: this.bundleUri.fsPath,
-      },
-    );
+    const basePaths = [
+      path.join(os.homedir(), ".rvm", "bin"),
+      "/usr/local/rvm/bin",
+      "/usr/share/rvm/bin",
```
[the remark this answers is not in the document]
Comment (onwardmk) [reply vscode/src/ruby/rvm.ts:25]: Hello, I opened #1862. I installed RVM on Ubuntu with [this package](https://github.com/rvm/ubuntu_rvm) per the instructions on the official RVM site. This package only installs RVM in `/usr/share/rvm`.

image

I can't speak to installing in `/usr/local/rvm`, but I hope that helps.

**Comment by vinistock** [comment]
I really appreciate the PR, but since this is affecting many users I ended up putting a fix together so that we can get it out there asap #1868.

Thanks again for contributing!

**Comment by joennlae** [comment]
> I really appreciate the PR, but since this is affecting many users I ended up putting a fix together so that we can get it out there asap #1868.
> 
> Thanks again for contributing!

Thank you for taking it over. I just already spent the time to fix it on my machine so I thought I could contribute :-)

I looked at your PR and it seems to me like a much better solution :-)

Just to answer the question to how I got to my path:

I installed rvm following the official instructions: (https://rvm.io/rvm/install)

```bash
curl -sSL https://get.rvm.io | bash
```

**Pull request closed without merging**